### ISSUE TITLE
Current state upgradability

### ISSUE THREAD

**Issue by tristan-cottam** [rancher/elemental#162]
Will Elemental, in its current state, be upgradable to future versions ? And if not, are there any estimates as to when it will be ?

**Comment by kkaempf** [comment]
Well, not in a strict sense. As we're following cloud-native principles, there's no upgrade, only replace/redeploy 😉 
But maybe I misunderstood your question 🤔

**Comment by tristan-cottam** [comment]
I don't have much knowledge in the repository structure and methodology being used.

What I mean is, Elemental is capable of upgrading to (or rather *being replaced by*, if I understand correctly), future versions of the operating system. Is Elemental currently capable of such ? Or is the operating system permanently immutable (which I doubt) ?

Maybe rewording it would help; if I install Elemental right now, in its current release, with rancherd and all, will it be, for the lack of better words, *future-proof* ? Will there be an clear upgrade path for, for instance, the replacement of rancherd by system-agent ?

**Comment by kkaempf** [comment]
The "rancherd" variant is/was experimental and there won't be a seamless switch to its successor, system-agent. You will need to adapt machineRegistration (and probably need to re-create machineInventory).
Once we do an official release of the stack, we will ensure upgradability to later versions.

**Comment by tristan-cottam** [comment]
Any idea when we could expect an official release, by any chance ?

**Comment by kkaempf** [comment]
By end of month 🤞🏻
